AIDevGallery/Samples/SharedCode/IChatClient/PhiSilicaClient.cs

```diff
@@ -219,53 +213,48 @@ public async IAsyncEnumerable<string> GenerateStreamResponseAsync(string prompt,
         string currentResponse = string.Empty;
         using var newPartEvent = new ManualResetEventSlim(false);
 
-        if (!_languageModel.IsPromptLargerThanContext(prompt))
+        IAsyncOperationWithProgress<LanguageModelResponseResult, string>? progress;
+        if (options == null)
         {
-            IAsyncOperationWithProgress<LanguageModelResponse, string>? progress;
-            if (options == null)
-            {
-                progress = _languageModel.GenerateResponseWithProgressAsync(new LanguageModelOptions(), prompt, new ContentFilterOptions(), _languageModelContext);
-            }
-            else
+            progress = _languageModel.GenerateResponseAsync(_languageModelContext, prompt, new LanguageModelOptions());
+        }
+        else
+        {
+            var modelOptions = GetModelOptions(options);
+            progress = _languageModel.GenerateResponseAsync(_languageModelContext, prompt, modelOptions);
+        }
```

Comment:
You could make GetModelOptions always return an object, then this becomes a single line.
If options == null, return new LanguageModelOptions()